src/gluonts/torch/model/deepar/_module.py

```diff
@@ -0,0 +1,323 @@
+            embedding_dims=self.embedding_dimension,
+        )
+        if scaling:
+            self.scaler = MeanScaler(dim=1, keepdim=True)
+        else:
+            self.scaler = NOPScaler(dim=1, keepdim=True)
+        self.lagged_rnn = LaggedLSTM(
+            input_size=1,  # TODO fix
+            features_size=self._number_of_features,
+            num_layers=num_layers,
+            hidden_size=hidden_size,
+            dropout_rate=dropout_rate,
+            lags_seq=[lag - 1 for lag in self.lags_seq],
+        )
+
+    @property
+    def _number_of_features(self) -> int:
+        return (
+            sum(self.embedding_dimension)
+            + self.num_feat_dynamic_real
+            + self.num_feat_static_real
+            + 1  # the log(scale)
+        )
+
+    @property
+    def _past_length(self) -> int:
+        return self.context_length + max(self.lags_seq)
+
+    def unroll_lagged_rnn(
+        self,
+        feat_static_cat: torch.Tensor,
+        feat_static_real: torch.Tensor,
+        past_time_feat: torch.Tensor,
+        past_target: torch.Tensor,
+        past_observed_values: torch.Tensor,
+        future_time_feat: Optional[torch.Tensor] = None,
+        future_target: Optional[torch.Tensor] = None,
+    ) -> Tuple[
+        Tuple[torch.Tensor, ...],
+        torch.Tensor,
+        Tuple[torch.Tensor, torch.Tensor],
+    ]:
+        context = past_target[:, -self.context_length :]
+        observed_context = past_observed_values[:, -self.context_length :]
+        _, scale = self.scaler(context, observed_context)
+
+        prior_input = past_target[:, : -self.context_length] / scale
+        input = (
+            torch.cat((context, future_target[:, :-1]), dim=1) / scale
+            if future_target is not None
+            else context / scale
+        )
+
+        unroll_length = (
+            self.context_length
+            if future_target is None
+            else self.context_length + future_target.shape[1] - 1
+        )
+        assert input.shape[1] == unroll_length
+
+        embedded_cat = self.embedder(feat_static_cat)
+        static_feat = torch.cat(
+            (embedded_cat, feat_static_real, scale.log()),
+            dim=1,
+        )
+        expanded_static_feat = static_feat.unsqueeze(1).expand(
+            -1, unroll_length, -1
+        )
+
+        time_feat = (
+            torch.cat(
+                (
+                    past_time_feat[:, -self.context_length + 1 :, ...],
+                    future_time_feat,
+                ),
+                dim=1,
+            )
+            if future_time_feat is not None
+            else past_time_feat[:, -self.context_length + 1 :, ...]
+        )
+
+        features = torch.cat((expanded_static_feat, time_feat), dim=-1)
+
+        output, new_state = self.lagged_rnn(prior_input, input, features)
+
+        params = self.param_proj(output)
+        return params, scale, static_feat, new_state
+
+    def output_distribution(
```

Comment:
I'm not sure if this function is fully JIT supported for all `torch.Distribution` objects. Mark it as `@torch.jit.ignore` otherwise.